flutter/flutter
# Reland "Add API to services package that overrides HTTP ban (#54243)"

## Description

This is being submitted in preparation of breaking change described in dart-lang/sdk#40548.

The Dart SDK support for this is added dart-lang/sdk@74a20b7.

Following this PR, I will draft and send out an announcement for flutter-announce@. Afterwards, I will submit the change to engine to ban HTTP by default on iOS and Android.

## Related Issues

flutter/flutter#54448

## Tests

I added the following tests:
- Make sure the symbol being used actually throws a [StateError] when HTTP protocol is used.
- Make sure we are setting the right value in the zone variables via `allowHttp`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.


## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] Yes, this is a breaking change. 


[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
